diff -up wrapland-0.601.0/server/wayland/resource.h.omv~ wrapland-0.601.0/server/wayland/resource.h
--- wrapland-0.601.0/server/wayland/resource.h.omv~	2024-08-23 20:34:31.773822642 +0200
+++ wrapland-0.601.0/server/wayland/resource.h	2024-08-23 20:34:41.213897999 +0200
@@ -47,7 +47,7 @@ public:
              wl_interface const* interface,
              void const* impl,
              Handle* handle)
-        : Resource(parent->client(), parent->version(), id, interface, impl, handle)
+        : Resource(parent->client, parent->version, id, interface, impl, handle)
     {
     }
 
